java-checks/src/main/java/org/sonar/java/symexecengine/ExecutionState.java

```diff
@@ -169,4 +188,31 @@ public void markValueAs(SymbolicValue value, State state) {
     stateOfValue.put(value, state);
   }
 
+  /**
+   * returns the merged state of all accessible values.
+   *
+   * @param symbol symbol whose merged state must be retrieved.
+   * @return merged state of all accessible values.
+   */
+  public State mergeAllReachableState(Symbol symbol) {
+    State result = null;
+    for (SymbolicValue value : getAllReachableValues(symbol)) {
+      State state = getStateOfValue(value);
+      result = result != null ? state.merge(result) : state;
```

Comment:
![New](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/new.png) ![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'state' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)